### PR TITLE
Fix opcode bugs and update script tests with ABC test cases

### PR DESCRIPTION
**Pull request by cpacia**
The current implementation differs from the ABC implementation in that we pop a
scriptnum off the stack (the NUM in NUM2BIN) but ABC pops a byte array off the
stack. This is a fix for it and a test.

This PR now also includes:
- Refactoring to use the full script test cases from ABC.
- ScriptVerifyMinimalIf and ScriptVerifyCompressedPubkey
- Bug fix in OP_CHECKMULTISIG and OP_CHECKMULTISIGVERIFY.

### CODE REVIEW

File: txscript/opcode.go [hunk 1]
```diff
@@ -2510,6 +2510,10 @@ func opcodeCheckMultiSig(op *parsedOpcode, vm *Engine) error {
 				return err
 			}
 
+			if vm.hasFlag(ScriptVerifySchnorr) && len(signature) == 64 {
```
Comment (markblundeberg) [inline txscript/opcode.go]: should be len(signature) == 65     (64 for the sig + 1 for hashtype byte).

File: txscript/opcode.go [hunk 2]
```diff
@@ -2510,6 +2510,10 @@ func opcodeCheckMultiSig(op *parsedOpcode, vm *Engine) error {
...
+				return scriptError(ErrSigInvalidDataLen, "CHECKMULTISIG cannot contain 64 byte signatures")
```
Comment (markblundeberg) [inline txscript/opcode.go]: Likewise in ABC we say "Signature cannot be 65 bytes in CHECKMULTISIG" here

File: txscript/opcode.go (hunk 1 above)
Comment (cpacia) [reply txscript/opcode.go]: Yeah right above it pops the sighash off so this will be 64 here.

File: txscript/scriptnum.go [hunk 3]
```diff
@@ -223,3 +223,56 @@ func makeScriptNum(v []byte, requireMinimal bool, scriptNumLen int) (scriptNum,
...
+
+	// If the last byte is not 0x00 or 0x80, we are minimally encoded.
+	last := data[len(data)-1]
+	if last&0x7f > 0 {
+		ret = make([]byte, len(data))
+		copy(ret, data)
+		return
+	}
+
+	// If the script is one byte long, then we have a zero, which encodes as an
```
Comment (markblundeberg) [inline txscript/scriptnum.go]: "If the script is one byte long," comment doesn't make sense but that's what ABC has lol

File: txscript/opcode.go (hunk 1 above)
Comment (markblundeberg) [reply txscript/opcode.go]: ah never mind, I see `signature` has hashtype already stripped off

**Comment by markblundeberg** [comment]
Looks good to me

**Pull request merged**